korean.Rmarkdown

```diff
@@ -0,0 +1,116 @@
+
+---
+title: Korean
+weight: 20
+
+{{% author %}}By Oul Han{{% /author %}} 
+
+```{r, message=FALSE}
+require(quanteda)
+require(quanteda.corpora)
+```
+
+## Tokenization
+
+You can use `tokens()` or a morphological analysis tool such as [KoNLP](https://cran.r-project.org/web/packages/KoNLP/index.html) to tokenize Korean texts. The sample corpus contains all 17 transcripts of commencement speeches by Korean presidents from 1948 to 2008. 
+
+```{r, eval=FALSE}
+corp <- download("data_corpus_korean-presidential-speeches")
+txt <- tail(texts(corp), 1000)
+```
+
+### Rule-based boundary detection
+
+`tokens()` can segment Korean texts in the same way as English texts based on white spaces, but cannot slice off many morphemes that are not necessary for text analysis. However, you may find that `tokens()` suffices, especially if supplemented with stop words that you tailor for the corpus in question.
+
+```{r include=FALSE}
+# This code is only for website generation
+corp <- readRDS("data_corpus_korean-presidential-speeches.RDS")
+txt <- tail(texts(corp), 1000)
+```
+
+```{r}
+icu_toks <- tokens(txt)
+head(icu_toks[[10]], 50)
+```
+
+### Morphological analysis 
+
+If you want to perform more accurate tokenization, you need to install a morphological analysis tool, and call it from R. [KoNLP](https://cran.r-project.org/web/packages/KoNLP/index.html) is one of the most popular morphological analysis tools for Korean texts. Using **KoNLP**'s *noun extraction* function, we can remove grammatical letters in a similar manner as lemmatization in English.
+
+```{r, message=FALSE}
+require(KoNLP)
+```
+
+```{r, eval=FALSE}
+konlp_toks <- download("data_list_korean-presidential-speeches.RDS")
+```
+
+```{r include=FALSE}
+# This code is only for website generation
+konlp_toks <- readRDS("data_list_korean-presidential-speeches.RDS")
+```
+
+```{r, eval=FALSE}
+konlp_toks <- KoNLP::extractNoun(txt)
+```
+
+```{r}
+head(konlp_toks[[10]], 50)
+```
+
+`konlp_toks` is a list object, but `as.tokens()` turns it into a `tokens` object.
+
+```{r}
+noun_toks <- as.tokens(konlp_toks)
+```
+
+```{r}
+head(noun_toks[[10]], 50)
+```
+
+### Refining tokens
+
+Sometimes, we should remove Chinese words that occur in old or technical Korean texts. In this sample corpus, they tend to appear in parentheses following their Korean phonetization. We will use regular expressions and the unicode character class ["Han"](http://unicode.org/faq/han_cjk.html) to remove Chinese characters that are either in, or not in, parentheses.
+
+```{r}
+head(noun_toks[[2]], 40)
+```
+
+In order to match parentheses, we have to escape them by `\\` because they are special characters in regular expression. `p{}` is also a regular expression to specify character class.
+
+```{r}
+just_korean <- gsub("\\(?\\p{Han}+\\)?", "", corp, perl = TRUE) 

```

Comment:
Is there any reason not doing?
```r
refi_noun_toks <- noun_toks
refi_pure_korean <- tokens_remove(refi_noun_toks, "\\(?\\p{Han}+\\)?", valuetype = "regex") 
```

Reply:
Is there any way to separate Korean and Chinese words in tokenization?

Reply:
It seems like KoNLP cannot. Alternative package (same tokenizer) `RmecabKo` didn't work.